cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/aws/aws-mc-common/aws-orchestrator/AwsOrchestrator.service.tsx

```diff
@@ -5,32 +5,55 @@ import { AwsDefaults } from '../default-service/AwsDefaults.service';
 import { STORE_SECTION_FORM } from '../../../../../state-management/reducers/Form.reducer';
 import { AWS_ADD_RESOURCES } from '../../../../../state-management/actions/Resources.actions';
 import { AwsResourceAction, FormAction, StoreDispatch } from '../../../../../shared/types/types';
+import { AWSKeyPair } from '../../../../../swagger-api/models/AWSKeyPair';
 interface AwsOrchestratorProps {
     awsState: { [key: string]: any };
     awsDispatch: StoreDispatch;
+    errorMessage: { [key: string]: any };
+    setErrorMessage: (newErrorMessage: { [key: string]: any }) => void;
 }
 export class AwsOrchestrator {
     static async initOsImages(props: AwsOrchestratorProps) {
-        clearOsImages(props);
-        const osImages = await AwsService.getAwsosImages(props.awsState[STORE_SECTION_FORM].REGION);
-        saveOsImages(props, osImages);
-        setDefaultOsImage(props, osImages);
+        const { awsState, setErrorMessage, errorMessage } = props;
+        try {
+            clearPreviousDiscoveryData(props, AWS_FIELDS.OS_IMAGE);
+            const osImages = await AwsService.getAwsosImages(awsState[STORE_SECTION_FORM].REGION);
+            saveCurrentDiscoveryData(props, AWS_FIELDS.OS_IMAGE, osImages);
+            setDefaultOsImage(props, osImages);
+            setErrorMessage(removeErrorInfo(errorMessage, AWS_FIELDS.OS_IMAGE));
+        } catch (e) {
+            setErrorMessage(addErrorInfo(errorMessage, e, AWS_FIELDS.OS_IMAGE));
+        }
+    }
+
+    static async initEC2KeyPairs(props: AwsOrchestratorProps, setKeyPairs: (keyPairs: AWSKeyPair[]) => void) {
+        const { awsState, setErrorMessage, errorMessage } = props;
+        try {
+            clearPreviousDiscoveryData(props, AWS_FIELDS.EC2_KEY_PAIR);
```

Comment:
should we move this line to catch block?

Reply:
Yes sir. Did not consider that previously. Moved it. Thansk